From e5759a0073954e43c770762e9c1d011dda0259e8 Mon Sep 17 00:00:00 2001
From: Bart Ribbers <bribbers@disroot.org>
Date: Fri, 25 Sep 2020 11:18:18 +0200
Subject: [PATCH] Include library version in libsurvive target

Without this linking to libsurvive would fail for e.g. Monado (at least
on Alpine Linux):
/usr/lib/gcc/x86_64-alpine-linux-musl/10.2.0/../../../../x86_64-alpine-linux-musl/bin/ld: cannot find -lsurvive
---
 CMakeLists.txt     | 4 +++-
 src/CMakeLists.txt | 5 ++++-
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f214f5e..6bd9639 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,8 @@
 cmake_minimum_required(VERSION 3.12.4)
 
-project(libsurvive C CXX)
+project(libsurvive
+	LANGUAGES C CXX
+	VERSION 0.2)
 include(CheckIncludeFile)
 include (CheckSymbolExists)
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index d32ae5d..167bf41 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -51,7 +51,10 @@ if(USE_HIDAPI OR WIN32)
 endif()
 
 add_library(survive ${SURVIVE_LIBRARY_TYPE} ${SURVIVE_SRCS})
-set_target_properties(survive PROPERTIES FOLDER "libraries")
+set_target_properties(survive PROPERTIES
+	FOLDER "libraries"
+	SOVERSION ${PROJECT_VERSION_MAJOR}
+	VERSION ${PROJECT_VERSION})
 
 target_link_libraries(survive minimal_opencv ${ADDITIONAL_LIBRARIES} mpfit)
 IF(HAVE_ZLIB_H)
